ci-operator/step-registry/cucushift/installer/check/vsphere/cpms/cucushift-installer-check-vsphere-cpms-commands.sh

```diff
@@ -0,0 +1,89 @@
+    else
+	echo "ERROR: For multi failure domains The network,workspace and template under cpms should be empty. network is ${cpms_network},workspace is ${cpms_workspace},template is ${cpms_template}" 
+        check_result=1
+    fi
+    
+#check the name of failuredomains in cpms should be same with install-config.
+    INSTALL_CONFIG="${SHARED_DIR}/install-config.yaml"
+    readarray -t zones_setting_from_config < <(yq-go r ${INSTALL_CONFIG} 'controlPlane.platform.vsphere.zones[*]')    
+    fd_name_cpms=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.failureDomains.vsphere[].name')
+    expected_fd_name=${zones_setting_from_config[*]}
+    if [[ ${fd_name_cpms} == "${expected_fd_name}" ]]; then
+	echo "INFO: The failure domain name are same between install_config and cmps"
+    else
+	echo "ERROR: The failure domain name are different between install_config and cmps. please check"
+        check_result=1
+    fi
+    return ${check_result}
+}
+
+
+function checkSinglezone() {
+        check_result=0
+	node_name=$(oc get machines.machine.openshift.io -n openshift-machine-api --selector machine.openshift.io/cluster-api-machine-type=master --no-headers | awk '{print $1}' | head -n1)
+	machine_workspace=$(oc get machine ${node_name} -n openshift-machine-api -ojson | jq -r '.spec.providerSpec.value.workspace')
+	machine_template=$(oc get machine ${node_name} -n openshift-machine-api -ojson | jq -r '.spec.providerSpec.value.template')
+	machine_network=$(oc get machine ${node_name} -n openshift-machine-api -ojson | jq -r '.spec.providerSpec.value.network.devices[].networkName')
+        
+	cpms_network=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.network.devices[].networkName')
+	cpms_workspace=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.workspace')
+        cpms_template=$(oc get controlplanemachineset -n openshift-machine-api -ojson | jq -r '.items[].spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.template')
+        if [[ ${machine_workspace} == "${cpms_workspace}" ]] && [[ ${machine_template} == "${cpms_template}" ]]  && [[ ${machine_network} == "${cpms_network}" ]]; then
+	    echo "INFO: The network,workspace and template setting between cmps and machines are same. that's expected"
+	else
+	   check_result=1
+	fi
+    return ${check_result}
+}
+
+function checkTemplate() {
+
+#check template in infrastructure should be populated.
+    check_result=0
+    infra_template=$(oc get infrastructure -n openshift-machine-api -ojson | jq -r '.items[].spec.platformSpec.vsphere.failureDomains[].topology.template')
```

Comment:
From the output in https://github.com/openshift/release/pull/48371#discussion_r1477261889, sounds no `spec.platformSpec.vsphere.failureDomains[].topology.template` key?

Reply:
Another suggestion:
```suggestion
    infra_template=$(oc get infrastructure -ojson | jq -r '.items[].spec.platformSpec.vsphere.failureDomains[].topology.template')
```

Reply:
> From the output in [#48371 (comment)](https://github.com/openshift/release/pull/48371#discussion_r1477261889), sounds no `spec.platformSpec.vsphere.failureDomains[].topology.template` key?
emmm, my bad, i copied the infra from a tp disabled cluster.
please refer this one:
oc get infrastructure -n openshift-machine-api -ojson
{
    "apiVersion": "v1",
    "items": [
        {
            "apiVersion": "config.openshift.io/v1",
            "kind": "Infrastructure",
            "metadata": {
                "creationTimestamp": "2024-02-01T01:34:35Z",
                "generation": 1,
                "name": "cluster",
                "resourceVersion": "526",
                "uid": "238b4e83-5cdd-4dc6-abef-53229d511fe9"
            },
            "spec": {
                "cloudConfig": {
                    "key": "config",
                    "name": "cloud-provider-config"
                },
                "platformSpec": {
                    "type": "VSphere",
                    "vsphere": {
                        "failureDomains": [
                            {
                                "name": "us-east-1",
                                "region": "us-east",
                                "server": "vcs8e-vc.ocp2.dev.cluster.com",
                                "topology": {
                                    "computeCluster": "/IBMCloud/host/vcs-mdcnc-workload-1",
                                    "datacenter": "IBMCloud",
                                    "datastore": "/IBMCloud/datastore/mdcnc-ds-shared",
                                    "networks": [
                                        "ci-vlan-1287"
                                    ],
                                    "resourcePool": "/IBMCloud/host/vcs-mdcnc-workload-1//Resources",
                                    "template": "/IBMCloud/vm/wweitest1-drglw-rhcos-us-east-us-east-1a"
                                },
                                "zone": "us-east-1a"
                            },
                            {
                                "name": "us-east-2",
                                "region": "us-east",
                                "server": "vcs8e-vc.ocp2.dev.cluster.com",
                                "topology": {
                                    "computeCluster": "/IBMCloud/host/vcs-mdcnc-workload-2",
                                    "datacenter": "IBMCloud",
                                    "datastore": "/IBMCloud/datastore/mdcnc-ds-shared",
                                    "networks": [
                                        "ci-vlan-1289"
                                    ],
                                    "resourcePool": "/IBMCloud/host/vcs-mdcnc-workload-2//Resources",
                                    "template": "/IBMCloud/vm/wweitest1-drglw-rhcos-us-east-us-east-2a"
                                },
                                "zone": "us-east-2a"
                            },
                            {
                                "name": "us-east-3",
                                "region": "us-east",
                                "server": "vcs8e-vc.ocp2.dev.cluster.com",
                                "topology": {
                                    "computeCluster": "/IBMCloud/host/vcs-mdcnc-workload-3",
                                    "datacenter": "IBMCloud",
                                    "datastore": "/IBMCloud/datastore/mdcnc-ds-shared",
                                    "networks": [
                                        "ci-vlan-1300"
                                    ],
                                    "resourcePool": "/IBMCloud/host/vcs-mdcnc-workload-3//Resources",
                                    "template": "/IBMCloud/vm/wweitest1-drglw-rhcos-us-east-us-east-3a"
                                },
                                "zone": "us-east-3a"
                            },
                            {
                                "name": "us-west-1",
                                "region": "us-west",
                                "server": "vcs8e-vc.ocp2.dev.cluster.com",
                                "topology": {
                                    "computeCluster": "/datacenter-2/host/vcs-mdcnc-workload-4",
                                    "datacenter": "datacenter-2",
                                    "datastore": "/datacenter-2/datastore/mdcnc-ds-shared",
                                    "networks": [
                                        "ci-vlan-1287"
                                    ],
                                    "resourcePool": "/datacenter-2/host/vcs-mdcnc-workload-4//Resources",
                                    "template": "/datacenter-2/vm/wweitest1-drglw-rhcos-us-west-us-west-1a"
                                },
                                "zone": "us-west-1a"
                            }
                        ],
                        "nodeNetworking": {
                            "external": {},
                            "internal": {}
                        },
                        "vcenters": [
                            {
                                "datacenters": [
                                    "IBMCloud"
                                ],
                                "port": 443,
                                "server": "vcs8e-vc.ocp2.dev.cluster.com"
                            }
                        ]
                    }
                }
            },
            "status": {
                "apiServerInternalURI": "https://api-int.wweitest1.qe.devcluster.openshift.com:6443",
                "apiServerURL": "https://api.wweitest1.qe.devcluster.openshift.com:6443",
                "controlPlaneTopology": "HighlyAvailable",
                "cpuPartitioning": "None",
                "etcdDiscoveryDomain": "",
                "infrastructureName": "wweitest1-drglw",
                "infrastructureTopology": "HighlyAvailable",
                "platform": "VSphere",
                "platformStatus": {
                    "type": "VSphere",
                    "vsphere": {
                        "apiServerInternalIP": "10.38.201.137",
                        "apiServerInternalIPs": [
                            "10.38.201.137"
                        ],
                        "ingressIP": "10.38.201.138",
                        "ingressIPs": [
                            "10.38.201.138"
                        ],
                        "loadBalancer": {
                            "type": "OpenShiftManagedDefault"
                        }
                    }
                }
            }
        }
    ],
    "kind": "List",
    "metadata": {
        "resourceVersion": ""
    }
}